packages/backend/src/open_payments/payment_pointer/service.test.ts

```diff
@@ -133,6 +134,112 @@ describe('Open Payments Payment Pointer Service', (): void => {
     })
   })
 
+  describe('Update Payment Pointer', (): void => {
+    test.each`
+      initialStatus                    | status                           | publicName
+      ${PaymentPointerStatus.Active}   | ${undefined}                     | ${'Public Name 1'}
+      ${PaymentPointerStatus.Active}   | ${PaymentPointerStatus.Inactive} | ${'Public Name 2'}
+      ${PaymentPointerStatus.Inactive} | ${PaymentPointerStatus.Active}   | ${null}
+      ${PaymentPointerStatus.Inactive} | ${undefined}                     | ${null}
+    `(
+      'Payment pointer status and publicName can be updated to $status and $publicName from initial status of: $initialStatus',
+      async ({ initialStatus, status, publicName }): Promise<void> => {
+        const paymentPointer = await createPaymentPointer(deps)
+
+        if (initialStatus === PaymentPointerStatus.Inactive) {
+          await paymentPointer.$query(knex).patch({ deactivatesAt: new Date() })
+        }
+
+        const updatedPaymentPointer = await paymentPointerService.update({
+          id: paymentPointer.id,
+          status,
+          publicName
+        })
+        assert.ok(!isPaymentPointerError(updatedPaymentPointer))
+
+        expect(updatedPaymentPointer.publicName).toEqual(publicName)
+
+        // check status. should be whatever the update input is, or the original value if not updated
+        let statusChecked = false
+        if (
+          status === PaymentPointerStatus.Inactive ||
+          (typeof status === 'undefined' &&
+            initialStatus === PaymentPointerStatus.Inactive)
+        ) {
+          expect(updatedPaymentPointer.deactivatesAt).toBeDefined()
+          expect(updatedPaymentPointer.isActive()).toEqual(false)
+          expect(updatedPaymentPointer.status).toEqual(
+            PaymentPointerStatus.Inactive
+          )
+          statusChecked = true
+        } else if (
+          status === PaymentPointerStatus.Active ||
+          (typeof status === 'undefined' &&
+            initialStatus === PaymentPointerStatus.Active)
+        ) {
+          expect(updatedPaymentPointer.deactivatesAt).toBe(null)
+          expect(updatedPaymentPointer.isActive()).toEqual(true)
+          expect(updatedPaymentPointer.status).toEqual(
+            PaymentPointerStatus.Active
+          )
+          statusChecked = true
+        }
+        // ensure all cases are checked
+        assert.ok(statusChecked)
+
+        await expect(
+          paymentPointerService.get(paymentPointer.id)
+        ).resolves.toEqual(updatedPaymentPointer)
+      }
+    )
+
+    test('Payment pointer fields can be updated independently', async (): Promise<void> => {
```

Comment:
I would break this test up into two tests/two cases

Reply:
i split these